### DISCUSSION TITLE
Projet de Boîtier pour Photobooth en Impression 3D - Raspberry Pi 4 ou 5 <-> 3D-Printed Photobooth Enclosure Project - Raspberry Pi 4 or 5

### DISCUSSION THREAD

**Discussion by cedric-Swiss** [photobooth-app/photobooth-app#369] [Show and tell]
![20241110_104317 - Copie](https://github.com/user-attachments/assets/48eae42a-eab4-4fdf-a107-9f85524e6d91)
![20241110_104329 - Copie](https://github.com/user-attachments/assets/21bfef2a-9ff5-4ba2-becb-af5c871f13f3)
![couv - Copie](https://github.com/user-attachments/assets/ecc80ecb-e76c-4334-b8b8-e7a151d2d3c5)

****************************** English version below in the text ****************************** 

Je vous présente mon projet de boîtier en impression 3D, conçu spécifiquement pour transformer un Raspberry Pi 4 en photobooth autonome et stylé. Ce boîtier est idéal pour des événements, des fêtes, ou tout simplement pour capturer des souvenirs avec un système DIY accessible et amusant !
Caractéristiques du boîtier :
•	Design élégant et compact : Ce boîtier est conçu pour être à la fois esthétique et fonctionnel, avec une structure légère mais robuste.
•	Adapté pour le Raspberry Pi 4 ou 5 : Avec des emplacements précis pour le Raspberry Pi, la caméra, l'écran tactile, et les ports d'alimentation.
•	Facile à assembler : Le boîtier est pensé pour simplifier l'assemblage, avec des pièces modulaires qui s'imbriquent facilement.
•	Compatibilité avec divers accessoires : Des supports sont inclus pour ajouter un éclairage LED et un déclencheur physique pour une meilleure expérience d'utilisation.
•	Organisation interne : Gestion des câbles intégrés, garantissant un intérieur propre et organisé (a peu près 😊).
Ce boîtier en impression 3D offre une solution simple et abordable pour créer votre propre photobooth. Que vous souhaitiez un photobooth pour des événements ou pour un projet de bricolage personnel.
Téléchargement et impression : Les fichiers STL sont disponibles en téléchargement et sont optimisés pour une impression facile en PLA ou PETG. Vous trouverez aussi des instructions d'assemblage détaillées pour garantir un montage sans soucis directement sur le lien suivant : https://makerworld.com/en/models/774344#profileId-710644
N’hésitez pas à partager vos impressions, vos versions personnalisées, ou des suggestions d’améliorations.

******************************  English version from here ****************************** 

3D-Printed Photobooth Enclosure Project - Raspberry Pi 4 or 5

I'm excited to share my 3D-printed enclosure project, specifically designed to turn a Raspberry Pi 4 or 5 into a sleek and standalone photobooth. This enclosure is perfect for events, parties, or simply capturing memories with an accessible and fun DIY setup!

Enclosure Features:

Elegant and Compact Design: This enclosure is both stylish and functional, with a lightweight yet sturdy structure.
Fitted for Raspberry Pi 4 or 5: Includes precise slots for the Raspberry Pi, camera, touchscreen, and power ports.
Easy Assembly: Designed for easy assembly with modular pieces that snap together effortlessly.
Compatible with Various Accessories: Supports for LED lighting and a physical trigger button are included to enhance the user experience.
Internal Cable Management: Integrated cable organization ensures a tidy interior (well, mostly 😊).
This 3D-printed enclosure offers a simple, affordable solution to create your own photobooth—whether for events or as a personal DIY project.

Download and Print: STL files are available for download and are optimized for easy printing in PLA or PETG. You'll also find detailed assembly instructions to ensure a smooth build at the following link: https://makerworld.com/en/models/774344#profileId-710644.

Feel free to share your prints, custom versions, or any suggestions for improvements!

**Comment by mgineer85** [comment]
Wow, this looks awesome :)  May I add some images and link your setup here: https://photobooth-app.org/projects/ ?

**Reply by cedric-Swiss** [reply]
Of course, no problem

**Reply by mgineer85** [reply]
How you control the buttons lights or is just static?

**Reply by mgineer85** [reply]
> Of course, no problem

Here we are:
https://photobooth-app.org/projects/#projet-de-boitier-pour-photobooth-en-impression-3d-made-in-switzerland

If you have any additions, just let me know :)

**Reply by cedric-Swiss** [reply]
> How you control the buttons lights or is just static?

Buttons are effectively static for the moment

However, for the led strips, I simply made this little script in python : 


```
#Script Arc-En-Ciel
import time
import board
from rainbowio import colorwheel
import neopixel

pixel_pin = board.D21
num_pixels = 27

pixels = neopixel.NeoPixel(pixel_pin, num_pixels, brightness=0.3, auto_write=False)


def color_chase(color, wait):
    for i in range(num_pixels):
        pixels[i] = color
        time.sleep(wait)
        pixels.show()
    time.sleep(0.5)


def rainbow_cycle(wait):
    for j in range(255):
        for i in range(num_pixels):
            rc_index = (i * 256 // num_pixels) + j
            pixels[i] = colorwheel(rc_index & 255)
        pixels.show()
        time.sleep(wait)


RED = (255, 0, 0)
YELLOW = (255, 150, 0)
GREEN = (0, 255, 0)
CYAN = (0, 255, 255)
BLUE = (0, 0, 255)
PURPLE = (180, 0, 255)

while True:
    pixels.fill(RED)
    pixels.show()
    # Vitesse de la couleur statique
    time.sleep(1)
    pixels.fill(GREEN)
    pixels.show()
    time.sleep(1)
    pixels.fill(BLUE)
    pixels.show()
    time.sleep(1)

    color_chase(RED, 0.1)  # Vitesse de la couleur changement
    color_chase(YELLOW, 0.1)
    color_chase(GREEN, 0.1)
    color_chase(CYAN, 0.1)
    color_chase(BLUE, 0.1)
    color_chase(PURPLE, 0.1)
```


I put this in a Rainbow.py file in my user folder (photobooth), then created a startup.sh file containing the following text: 


```
cd /home/Photobooth/
sudo python Rainbow.py
```


I then chmod u+x the startup.sh file

And I added in the crontab : 

`sudo crontab -e`

the following line: 

`@reboot sh /home/photobooth/startup.sh > /home/photobooth/logs/log.txt 2>&1`

and so each time you reboot, it activates the LEDs on the front panel.

**Reply by mgineer85** [reply]
Do you use the WLED integration of the app also?   I think it should be able to drive multiple neopixels and that it could be combined in one playlist so there is some activity when the app is in countdown, shoot, standby.

**Reply by cedric-Swiss** [reply]
No, I haven't used it yet.

I didn't want to add ESP to my setup.

If the WLED part allowed me to choose a GPIO as output, I'd be interested.

For the moment, I'm only using the led ribbon to decorate the BOX.

**Comment by cedric-Swiss** [comment]
Hello,

Thank you for your message.

I've added a few printscreens in the comments on the site where I make
the files available.

Unfortunately, I forgot to take photos during assembly.

I'll do some when I reopen it.

Best regards

**Reply by Striker62** [reply]
Hi , thank you for your responce. After I posted this Message I found your screen shots at Bamboo.

**Comment by Bartiiee** [comment]
Late to the party!
Which screen did you use?

**Reply by cedric-Swiss** [reply]
Hi,

I am using the following screen: 1 RB-LCD7-3 Touch Screen.

All information about the equipment used can be found at the link
provided.

See you soon.

Cédric

**Comment by Striker62** [comment]
Hi, i Use this   
Waveshare 7inch HDMI Control Panel Display Capacitive Touch LCD 1024×600 Resolution IPS Screen Latest Version Compatible with Raspberry Pi 5/Pi 4/3B+/Zero W/Mini PC, Monitor 7 Zoll

**Comment by nicolasborn** [comment]
Hi cedric
I'd like to build a Photo Booth using your great design.
Before I 3d print everything, do you still support it actively? I might need some help with the wiring of the leds, buttons or the cabling for the Neutrik powercons. Or is there someone else who already built it and can help me?
Regards from Bern,
Nicolas
